pilot/pkg/serviceregistry/kube/controller/endpoint_builder.go

```diff
@@ -46,14 +46,17 @@ type EndpointBuilder struct {
 	hostname string
 	// If specified, the fully qualified Pod hostname will be "<hostname>.<subdomain>.<pod namespace>.svc.<cluster domain>".
 	subDomain string
+	// The name of the node where the endpoint(pod) is located
+	NodeName string
 }
 
 func NewEndpointBuilder(c controllerInterface, pod *v1.Pod) *EndpointBuilder {
-	locality, sa, namespace, hostname, subdomain := "", "", "", "", ""
+	locality, sa, namespace, hostname, subdomain, nodeName := "", "", "", "", "", ""
 	var podLabels labels.Instance
 	if pod != nil {
 		locality = c.getPodLocality(pod)
 		sa = kube.SecureNamingSAN(pod)
+		nodeName = string(c.Cluster()) + "-" + pod.Spec.NodeName
```

Comment:
I think here you donot need add clusterID, it can be inferred from controllerInterface

Reply:
> I think here you donot need add clusterID, it can be inferred from controllerInterface

endpoint does not contains ClusteID，so add clusterID in nodeName
https://github.com/ChenglongGao327/istio/blob/bb131f5f88a881975565d2aa10d723b1776f0157/pilot/pkg/xds/endpoint_builder.go#L314

Reply:
nit: if we do this, can we make the delimiter `/`? Right now you can get conflicts between cluster names and node names still

Reply:
> nit: if we do this, can we make the delimiter `/`? Right now you can get conflicts between cluster names and node names still
i will change '-' to '/'